### DISCUSSION TITLE
airflow.exceptions.AirflowException: No log_template entry found for ID None.

### DISCUSSION THREAD

**Discussion by zcattacz** [apache/airflow#29382] [Q&A]
### Apache Airflow version

2.5.1

### What happened

I was really new to airflow weeks ago (on 2.5.0 atm), installed airflow for local db sync task testing without celery/rapidmq.
Airflow is setup with local postgresql 14 instance in schema `airflow`. 

I started with simple steps. With SequentialExecutor, tutorial task ran fine, then I decided use LocalExecutor. 
Things suddenly fell apart, the `ID None` exception starts to pops up, the tutorial dag simply fails all the time.
But since the url logs flood anything else away really quick I didn't pay enough attention.
At the same time, I saw the warning `Only works with the Celery, CeleryKubernetes or Kubernetes executors`
on webUI when debugging in the diagram view, 
Then all my attention is diverted to running it with CeleryExecutor and rapidmq etc.
As you can expect, even with rapidmq + celeryexecutor in place, things fail as usual, only SequentialExecutor works.
`airflow tasks test tutorial` always works during the whole process.

Since I am new, with every switching between executors the database is deliberately dropped to reset the setup.
As I run down the settings one by one, I decided to not use `schema`. Then suddenly everything works as usual, even with LocalExecutor.

The `ID None` behavior also applies to 2.5.1

### What you think should happen instead

LocalExecutor/CeleryExectuor should run fine like SequentialExecutor without `ID None` error.

### How to reproduce

Install apache-airflow with contraint.txt for py3.9 into a venv folder
run `airflow standalone`, stop it, 
cp $pyvenv/airflow/lib/python3.9/site-packages/airflow/example_dags/tutorial.py ~/airflow/dags/
edit `~/airflow/airflow.cfg`, 
set executor= LocalExecutor
set database= `postgresql+psycopg2://postgres:postgres_passwd@127.0.0.1:5432/testdb1?application_name=airflow`
set sql_alchemy_schema=airflow1
set load_example=False
set load_default_connection to false
sudo -u postgres; \c testdb1; create schema airflow1; \q
restart airflow

### Operating System

Debian 11.6

### Versions of Apache Airflow Providers

(airflow) user@localhost:~/repos/apach-airflow$ pip freeze|grep airflow
apache-airflow==2.5.1
apache-airflow-providers-common-sql==1.3.2
apache-airflow-providers-ftp==3.3.0
apache-airflow-providers-http==4.1.0
apache-airflow-providers-imap==3.1.1
apache-airflow-providers-sqlite==3.3.1

### Deployment

Other

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [ ] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by Cnidarias** [comment]
Hi - I ran into a very similar issue today.

You are telling airflow to use the `airflow` schema but your database user does not have the `airflow` schema in it's search path. You can either alter your postgres user to include the `airflow` schema in your search path or you can change your SQLAlchemy connection string to include the `-csearch_path` configuration option

Here is the section in the Airflow docs describing this [documentation](https://airflow.apache.org/docs/apache-airflow/stable/howto/set-up-database.html#other-configuration-options)

So for what you have in your connection string right now  you would have to adjust it like so

`postgresql+psycopg2://postgres:postgres_passwd@127.0.0.1:5432/testdb1?application_name=airflow&options=-csearch_path%3Dairflow`
